RIOT-OS/RIOT
# makefiles/suit: place keys in $XDG_DATA_HOME

### Contribution description

Placing the SUIT key in the RIOT repository folder is dangerous as a repo checkout is by most people considered a volatile location.
Since all important files are stored in git, deleting the entire folder or it's contents is not an uncommon cleanup operation.

If the user is at that point unaware that SUIT key material is stored in that folder, that key will then be lost.

Another workflow may involve multiple checkouts of the RIOT repository to multiple folders to work on several features at the same time, or for easy cross-referencing or splitting of off features from an integration into a feature branch.
In that case each checkout would use it's own incompatible SUIT key.

To avoid all these pitfalls, place the SUIT keys outside the RIOT repository in the `$XDG_DATA_HOME` directory.

### Testing procedure

    make -C examples/suit_update suit/publish

should still work (with a new key being generated in `~/.local/share/RIOT/keys`).

### Issues/PRs references
